packages/mdc-snackbar/mdc-snackbar.scss

```diff
@@ -21,152 +21,180 @@
+  // Ignore mouse events on the root layout element.
+  pointer-events: none;
 
-  @media (max-width: ($mdc-snackbar-tablet-breakpoint - 1)) {
-    bottom: 0;
-    left: 0;
-    width: 100%;
-    transform: translate(0, 100%);
-  }
+  // Mobile Safari only supports `rgba` values for this property; named values like `transparent` are ignored.
+  // From Apple's docs:
+  // > This property obeys the alpha value, if specified.
+  // > If you don’t specify an alpha value, Safari on iOS applies a default alpha value to the color.
+  // > To disable tap highlighting, set the alpha value to 0 (invisible).
+  // > If you set the alpha value to 1.0 (opaque), the element is not visible when tapped.
+  // See https://github.com/ben-eb/postcss-colormin/issues/1
+  -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
 }
 
-.mdc-snackbar--active {
-  transform: translate(0);
-  transition: mdc-animation-enter(transform, .25s);
-  pointer-events: auto;
-
-  &:not(.mdc-snackbar--align-start) {
-    transform: translate(-50%, 0);
-
-    @media (max-width: ($mdc-snackbar-tablet-breakpoint - 1)) {
-      bottom: 0;
-      left: 0;
-      width: 100%;
-      transform: translate(0);
-    }
-  }
+@at-root {
+  @include mdc-snackbar-surface-fill-color($mdc-snackbar-surface-fill-color);
+  @include mdc-snackbar-overlay-fill-color($mdc-snackbar-overlay-fill-color);
+  @include mdc-snackbar-label-ink-color($mdc-snackbar-label-ink-color);
+  @include mdc-snackbar-min-width($mdc-snackbar-min-width);
+  @include mdc-snackbar-max-width($mdc-snackbar-max-width);
+  @include mdc-snackbar-elevation($mdc-snackbar-elevation);
+  @include mdc-snackbar-shape-radius($mdc-snackbar-shape-radius);
 }
 
-.mdc-snackbar__action-wrapper {
-  @include mdc-rtl-reflexive-property(padding, 24px, 0);
+.mdc-snackbar--wide {
+  @include mdc-snackbar-viewport-margin($mdc-snackbar-viewport-margin-wide);
 }
 
-.mdc-snackbar--action-on-bottom {
-  flex-direction: column;
+.mdc-snackbar--leading {
+  justify-content: flex-start;
 }
 
-.mdc-snackbar__text {
-  @include mdc-typography(body1);
-  @include mdc-rtl-reflexive-box(margin, right, auto, ".mdc-snackbar");
+.mdc-snackbar--open,
+.mdc-snackbar--closing {
+  transform: none;
+}
 
+.mdc-snackbar__surface {
   display: flex;
+
+  // Non-static positioning is required to make `overflow: hidden` work in Safari.
+  // See https://bugs.webkit.org/show_bug.cgi?id=98538#c5
+  position: relative;
   align-items: center;
-  height: 48px;
-  transition: mdc-animation-exit-permanent(opacity, .3s);
+  justify-content: flex-start;
+  box-sizing: border-box;
+  transform: scale(.8);
   opacity: 0;
-  color: $mdc-snackbar-foreground-color;
 
-  @media (min-width: $mdc-snackbar-tablet-breakpoint) {
-    @include mdc-rtl-reflexive-property(padding, 0, 24px);
+  // Prevent mdc-snackbar__overlay's background-color from bleeding past the surface's border-radius.
+  overflow: hidden;
+
+  .mdc-snackbar--stacked & {
+    flex-direction: column;
+    align-items: flex-start;
```

Comment:
Should this flip for RTL?

Reply:
Nope! That's the beauty of flexbox and `flex-start`: It's already RTL-aware! 🎉 

BTW, you can add `?dir=rtl` to any screenshot test page URL to see what it looks like in RTL:

https://storage.googleapis.com/mdc-web-screenshot-tests/advorak/2018/11/09/02_51_51_193/spec/mdc-snackbar/index.html?dir=rtl

#### LTR:

![image](https://user-images.githubusercontent.com/409245/48295877-2b971c80-e445-11e8-990e-4dec45f285c1.png)

#### RTL:

![image](https://user-images.githubusercontent.com/409245/48295862-0bfff400-e445-11e8-97fc-d62c2770903e.png)